### PR TITLE
Clean up dns parsing and verification

### PR DESCRIPTION
**Pull request by osowskit**
* Use 'public_suffix' to parse DNS domain
* Simplify verification

TODO:
- [ ] test for edge cases (top level, non-standard suffix, etc)

### CODE REVIEW

File: dnsimple_hook.rb [hunk 1]
```diff
@@ -13,25 +14,20 @@
...
+    if dns_simple_domain_name.eql? domain_name
+      puts 'verified domain control'
+      return
     end
   end
   
-  # if you don't can't control the domain, then no need to go on further
+  # Stop execution of this script if the API key can't control
+  # this domain
   exit
```
Comment (jbjonesjr) [inline dnsimple_hook.rb]: for the record, I don't know if this works. we need to test that as well.

File: dnsimple_hook.rb [hunk 2]
```diff
@@ -75,11 +71,18 @@ def delete_dns(domain, txt_challenge)
...
-        // part of me wants to replace this with a much simpler method
-        domain_hash = find_domain(account_id, full_domain_name)
 
-        setup_dns(account_id, domain_hash["domain_name"], domain_hash["subdomain_name"] , txt_challenge)
+        # Split domain for DNSimple API
+        ps_domain = PublicSuffix.domain(full_domain_name)
+        domain_name = ps_domain.domain
+        subdomain_name = ps_domain.trd
+
+        # Search for domain in DNSimple or stop script
```
Comment (jbjonesjr) [inline dnsimple_hook.rb]: s/Search for/Confirm control of/

**Review by jbjonesjr — COMMENTED** [review]
Script looks great to me! Minor edit on the comment, and general note. I'd :ship: it and we can keep working in the main PR
